eng/references.targets

```diff
@@ -42,17 +65,23 @@
     </ItemGroup>
   </Target>
 
-  <!-- Make shared framework assemblies not app-local (non private). -->
-  <Target Name="UpdateProjectReferencesWithPrivateAttribute"
-          AfterTargets="AssignProjectConfiguration"
-          BeforeTargets="PrepareProjectReferences"
+  <!--
+    Make shared framework assemblies not app-local (non private), don't reference them and flow
+    the TargetFramework-TargetOS touple to avoid tfm selection.
+  -->
+  <Target Name="UpdateSharedFrameworkProjectReferences"
+          BeforeTargets="ResolveP2PReferences;AssignProjectConfiguration"
           Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
                      ('$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true') and
-                     '@(ProjectReferenceWithConfiguration)' != ''">
+                     '@(ProjectReference)' != ''">
     <ItemGroup>
-      <ProjectReferenceWithConfiguration PrivateAssets="all"
-                                         Private="false"
-                                         Condition="$(NetCoreAppLibrary.Contains('%(Filename);'))" />
+      <ProjectReference Condition="$(NetCoreAppLibrary.Contains('%(Filename);'))">
+        <PrivateAssets>all</PrivateAssets>
+        <Private>false</Private>
+        <ReferenceOutputAssembly Condition="!$(NetCoreAppLibraryNoReference.Contains('%(Filename);')) and
+                                            '%(ProjectReference.SkipUseReferenceAssembly)' != 'true'">false</ReferenceOutputAssembly>
+        <ReferringTargetFramework Condition="'$(TargetFrameworkSuffix)' == ''">$(TargetFramework)-$(TargetOS)</ReferringTargetFramework>
```

Comment:
This isn't handling the case when referencing project is a less specific RID than referenced.  I don't think you need this condition.

Reply:
> Nor does it handle the case where TFM is less specific (though we may not use this any more).

Right, we don't care about that anymore as test projects targeting ie .NETStandard aren't supported.

> This isn't handling the case when referencing project is a less specific RID than referenced.

Right, but tbh I don't understand your proposal below. `BuildTargetFramework` is the same as `TargetFramework` if the project is invoked individually (outside of a root build).

Reply:
> This isn't handling the case when referencing project is a less specific RID than referenced.

I thought about this a bit more and I'm unsure how we would make this ever work. I'm ok with supporting a rid-less `TargetFramework` for the sake of less configurations but I can't get my head around how we would support less specific RIDs without loosing the ability in test projects to not support some configurations.

Maybe we just chat about his offline as this confuses me... In the meantime I adjusted the two affected test projects for the sake of getting this in.

Reply:
Binplacing will still  make the right decision.  The purpose of the project reference here is just to make a convenient build.  That convenient build shouldn't really be chosen based on the test's configuration, but what the vertical build would have put in the test runtime, which is decided by the things I suggested.